code/common/analyticslib.q

```diff
@@ -48,10 +48,15 @@ ffill:{[args]
 / - General pivot function
 pivot:{[args]
 	checkargs[args;`table`by`piv`var;(" sS")];
- 	/ - if user has not specified f or g set to defaults
- 	if[not all `f`g in key args;
-    	args[`f]:{[v;P] `$"_" sv' string (v,()) cross P};
-    	args[`g]:{[k;P;c] k,asc c}];
+ 	/ - Check for optional args f and g
+	/ - Check if f optional exists, if empty set as default else leave as input
+	if[not `f in key args;
+	args[`f]:{[v;P] `$"_" sv' string (v,()) cross P};
+	args[`f]];
```

Comment:
Don't think the final line here is necessary/doing anything? Think this can just be

```suggestion
	if[not `f in key args;
	  args[`f]:{[v;P] `$"_" sv' string (v,()) cross P};
	];
```

(also good to indent the code within an `if` statement for readability)

Obviously same applies for `g` as well